source/aws-bootstrap-kit/lib/account.ts

```diff
@@ -84,72 +90,114 @@ export class Account extends Construct {
   readonly accountName: string;
   readonly accountId: string;
   readonly accountStageName?: string;
+  readonly accountStageOrder?: number;
 
   constructor(scope: Construct, id: string, accountProps: IAccountProps) {
     super(scope, id);
 
     const accountProvider = AccountProvider.getOrCreate(this);
 
-    let account = new core.CustomResource(
-      this,
-      `Account-${accountProps.name}`,
-      {
-        serviceToken: accountProvider.provider.serviceToken,
-        resourceType: "Custom::AccountCreation",
-        properties: {
-          Email: accountProps.email,
-          AccountName: accountProps.name,
-          AccountType: accountProps.type,
-          StageName: accountProps.stageName,
-          StageOrder: accountProps.stageOrder?.toString(),
-          HostedServices: accountProps.hostedServices?accountProps.hostedServices.join(':'):undefined
+    let existingAccount = false;
+    let accountId = accountProps.id;
+    let hostedServices = accountProps.hostedServices ? accountProps.hostedServices.join(':') : undefined;
+
+    // do not create account if we reuse one
+    let account;
+    if (!accountId) {
+      account = new core.CustomResource(
```

Comment:
Did you also had the idea to use the SNS topic and using a very custom state machine? So we can asynchronously create the account, then move and describe (maybe tag) in a single provider to reduce the number of Cloudformation resources.

How likely is it that the create account status will take longer than 15 minutes?

Reply:
You are right, in the case you create a single account with your payment method (credit card), the activation can take more than that (to validate the payment method). But as we are in an organization, it should not happen, and the account creation should take a few seconds / minutes only. 

Also, moving to an asynchronous architecture is quite a huge change. Not sure we need to go that far.